### ISSUE TITLE
Disable nspkg.pth magic when PEP-420 is available

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools#805]
In pypa/pip#1924, we find that setuptools is generating the `-nspkg.pth` files that are bootstrapping the namespace packages, but these files are unnecessary when PEP-420 is present (Python 3.3+) and in fact interfere with pure PEP-420 namespace packages.

It seems the solution may be simple - bypass this behavior when PEP-420 is detected.

**Comment by jaraco** [comment]
Turns out the implementation is pretty straightforward. In the 'issue-805' branch, I've committed probably the minimal changes to support this functionality. I've tagged and released these changes as 28.1.0b1, a wheel that I've [hosted here](https://github.com/pypa/setuptools/releases/download/v28.1.0b1/setuptools-28.1.0b1-py2.py3-none-any.whl). I've confirmed the technique works. I'm going to run with this version of setuptools for a while and see if I encounter any issues.

I invite others to try it out as well. Just:

```
pip install https://github.com/pypa/setuptools/releases/download/v28.1.0b1/setuptools-28.1.0b1-py2.py3-none-any.whl
```
